src/router.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash 
+
+# Common pipeline build script will 
+# delegate to appropriate composer/fabric build script
+# script receives two parameters, the stage to route  
+# and the platform 
+
+stage="$1"
+platform="$2"
```

Comment:
The routing should be automatic, not specified as a script argument
Simon's simple logic is here for example... https://github.com/IBM-Blockchain-Starter-Kit/build-lib/blob/master/scripts/pipeline-DEPLOY.sh#L24

Reply:
Actually we made it purposely this way for several reasons.  First it makes the code cleaner, having code figuring this out on it own is unnecessary.  The stages on the toolchain pipeline know exactly what they need to route to so when we call the scripts from the stages we will pass the parameters that are needed.   The other reason we purposely did it this way is because we wanted to limit as much as possible having logic in this repository that would depend on assumptions made in other repositories.  When Simon did that we had toolchain, scripts and scaffolding all together but now that we have split this into different parts we have to be careful not to be injecting logic on each repo that would couple us to the other repos.

Reply:
If there is no logic switching between platforms automatically, then I'm not sure what having a top level script is adding- you'll need to hard code the setting in the toolchain template (so it's no longer common) which is just adding complexity when you could specify which platform specific script to use directly.

Reply:
The use of the top level script is still the same,  having the top level script makes it possible to reuse the same toolchain for both composer and  go-chaincode.

Reply:
Let's  cover this in our conversation.